src/javaRosa.js

```diff
@@ -1452,9 +1331,10 @@ define([
         loadXML: function (xmlString) {
             var _this = this,
                 langs = this.opts().javaRosa.langs,
-                Itext;
+                Itext, itextMap;
 
             this.data.javaRosa.Itext = Itext = new ItextModel();
+            this.data.javaRosa.itextMap = itextMap = {};
```

Comment:
This temporary mapping of itext IDs to itext items is constructed during XML parsing, and used below in `populateControlMug`, which is called by the parser in the super call. The mapping is discarded once the form has been parsed. One possible issue is that itext from the XForm that is not referenced by a mug is discarded early (just after parsing). Previously it did not get discarded until the form was saved (see code removed from [`beforeSerialize`](https://github.com/dimagi/Vellum/pull/360/files#diff-e5b3eca8407b45b132e0147b8392192cL1643)).